x-pack/plugins/lens/public/xy_visualization/xy_config_panel.tsx

```diff
@@ -191,25 +234,72 @@ export function XyToolbar(props: VisualizationToolbarProps<State>) {
       <EuiFlexItem>
         <EuiFlexGroup gutterSize="none" responsive={false}>
-          <TooltipWrapper
-            tooltipContent={i18n.translate('xpack.lens.xyChart.fittingDisabledHelpText', {
-              defaultMessage: 'This setting only applies to line and area charts.',
+          <ToolbarPopover
+            title={i18n.translate('xpack.lens.xyChart.valuesLabel', {
+              defaultMessage: 'Values',
             })}
-            condition={!hasNonBarSeries}
+            type="values"
+            groupPosition="left"
+            buttonDataTestSubj="lnsMissingValuesButton"
           >
-            <ToolbarPopover
-              title={i18n.translate('xpack.lens.xyChart.valuesLabel', {
-                defaultMessage: 'Values',
-              })}
-              isDisabled={!hasNonBarSeries}
-              type="values"
-              groupPosition="left"
-              buttonDataTestSubj="lnsMissingValuesButton"
+            <EuiFormRow
+              display="columnCompressed"
+              label={
+                <TooltipWrapper
+                  tooltipContent={i18n.translate(
+                    'xpack.lens.xyChart.valuesStackedDisabledHelpText',
+                    {
+                      defaultMessage:
+                        'This setting can only be used on categorical, un-stacked bar charts',
+                    }
+                  )}
+                  condition={isValueLabelsDisabled}
+                >
+                  <span>
+                    {i18n.translate('xpack.lens.shared.chartValueLabelVisibilityLabel', {
+                      defaultMessage: 'Labels',
+                    })}{' '}
+                    {isValueLabelsDisabled ? (
+                      <EuiIcon
+                        type="questionInCircle"
+                        color="subdued"
+                        size="s"
+                        className="eui-alignTop"
+                      />
+                    ) : null}
+                  </span>
+                </TooltipWrapper>
+              }
             >
+              <EuiButtonGroup
+                isFullWidth
+                legend={i18n.translate('xpack.lens.shared.chartValueLabelVisibilityLabel', {
+                  defaultMessage: 'Labels',
+                })}
+                isDisabled={isValueLabelsDisabled}
+                data-test-subj="lnsValueLabelsDisplay"
+                name="valueLabelsDisplay"
+                buttonSize="compressed"
+                options={valueLabelsOptions}
+                idSelected={
+                  valueLabelsOptions.find(({ value }) => value === valueLabelsVisibilityMode)!.id
+                }
+                onChange={(modeId) => {
+                  const newMode = valueLabelsOptions.find(({ id }) => id === modeId)!.value;
+                  setState({ ...state, valueLabels: newMode });
+                }}
+              />
+            </EuiFormRow>
```

Comment:
It sounds we're likely back to square one here. :D
There's a plan to make value labels visible for all types of bars in the future: the way we handle this transition period (until the charting library supports more options or there's a decision to enable them as is) it's taking very long time to settle to a single UI, I think.

I do not have strong opinion here: it started with the button disabled and then enabled by request, so I do not mind either way.

Reply:
I think we struggled with the whole hiding vs. disabling in a bunch of other places as well. What do y'all think about merging this PR to avoid ongoing maintenance costs, come to a conclusion for Lens in general, then making everything consistent in a separate PR? This is how we handled lots of ongoing changes in the UI previously a few times.

Reply:
@flash1293: I'm fine with whatever approach you all deem best (either in this PR or a separate PR in the future). To be clear though, my concern is not about consistency.

@dej611: If the plan is to make the value labels option available for all bar-type visualizations in the future, then I would be fine with this form field being present whenever any bar-type visualization is selected, and simply disabling for now if it happens to be a stacked bar or histogram.

That said, what I would continue to find problematic is that this form field currently shows even when a non-bar visualization is selected. For example, this option continues to show for me (in a disabled state) when I've selected line, area, etc. Since a user can never interact with it for those non-bar visualization types, could we not hide it?